homeassistant/components/samsungtv/media_player.py

```diff
@@ -136,32 +138,55 @@ def __init__(self, host, port, name, timeout, mac, uuid):
             "name": "HomeAssistant",
             "description": name,
             "id": "ha.component.samsung",
+            "method": None,
             "port": port,
             "host": host,
             "timeout": timeout,
         }
 
+        # Select method by port number, mainly for fallback
         if self._config["port"] in (8001, 8002):
             self._config["method"] = "websocket"
-        else:
+        elif self._config["port"] == 55000:
             self._config["method"] = "legacy"
 
-    def update(self):
+    async def async_update(self):
         """Update state of device."""
         self.send_key("KEY")
 
     def get_remote(self):
         """Create or return a remote control instance."""
+
+        # Try to find correct method automatically
+        if self._config["method"] not in METHODS:
+            for method in METHODS:
+                try:
+                    self._config["method"] = method
+                    LOGGER.debug("try config: %s", self._config)
+                    self._remote = self._remote_class(self._config.copy())
+                    self._state = STATE_ON
+                    LOGGER.info("found working config: %s", self._config)
```

Comment:
We start logging messages with capital letter.

Reply:
Debug max.

Reply:
> Debug max.

What do you mean? Only use debug messages?

Reply:
Yes, please use debug instead of info level. This kind of logging message is only useful when debugging.